trieste/utils/misc.py

```diff
@@ -220,21 +233,84 @@ def map_values(f: Callable[[U], V], mapping: Mapping[K, U]) -> Mapping[K, V]:
 """ An unbound type variable. """
 
 
-def get_value_for_tag(mapping: Optional[Mapping[Tag, T]], tag: Tag = OBJECTIVE) -> Optional[T]:
-    """Return the value of a tag in a mapping.
+def get_value_for_tag(
+    mapping: Optional[Mapping[Tag, T]], tags: Union[Tag, Sequence[Tag]] = OBJECTIVE
+) -> Tuple[Optional[Tag], Optional[T]]:
+    """Return the value from a mapping for the first tag found from a sequence of tags.
 
     :param mapping: A mapping from tags to values.
-    :param tag: A tag.
-    :return: The value of the tag in the mapping, or None if the mapping is None.
-    :raises ValueError: If the tag is not in the mapping and the mapping is not None.
+    :param tags: A tag or a sequence of tags. Sequence is searched in order.
+    :return: The chosen tag and value of the tag in the mapping, or None for each if the mapping is
+        None.
+    :raises ValueError: If none of the tags are in the mapping and the mapping is not None.
     """
 
+    if isinstance(tags, Tag):
+        tags = [tags]
+
     if mapping is None:
-        return None
-    elif tag in mapping.keys():
-        return mapping[tag]
+        return None, None
     else:
-        raise ValueError(f"tag '{tag}' not found in mapping")
+        matched_tags = sorted(set(tags) & set(mapping.keys()), key=tags.index)
```

Comment:
Clearer and more efficient?

```suggestion
        matched_tag = next(tag for tag in tags if tag in mapping.keys(), None)
        if matched_tag is None:
            raise ValueError(...)
        return matched_tag, mapping[matched_tag]
```